rosidl_runtime_c/include/rosidl_runtime_c/message_bounds_struct.h

```diff
@@ -28,21 +28,50 @@ typedef struct rosidl_message_bounds_t rosidl_message_bounds_t;
 typedef const rosidl_message_bounds_t * (* rosidl_message_bounds_handle_function)(
   const rosidl_message_bounds_t *, const char *);
 
+/// Contains rosidl message bounds data
 struct rosidl_message_bounds_t
 {
+  /// String identifier for the type_support.
   const char * typesupport_identifier;
+  /// pointer to type support handle function
   const void * data;
+  /// pointer to type support handle function.
   rosidl_message_bounds_handle_function func;
 };
 
+/// Get the message bounds handle specific to this identifier.
+/**
+ * If the identifier is the same as this handle's identifier, then the handle is
+ * simply returned, otherwise it returns zero.
+ *
+ * \param handle Handle to service type support
+ * \param identifier The identifier to get the handle for
+ * \return The associated message bounds handle if found, otherwise NULL
+ */
 ROSIDL_GENERATOR_C_PUBLIC
 const rosidl_message_bounds_t * get_message_bounds_handle(
   const rosidl_message_bounds_t * handle, const char * identifier);
 
+/// Get the message bounds handle function specific to this identifier.
+/**
+ * If the identifier is the same as this handle funtion's identifier, then the handle is
+ * simply returned, otherwise it returns zero.
+ *
+ * \param handle Handle to service type support
+ * \param identifier The identifier to get the handle function for
+ * \return The associated message bounds handle function if found, otherwise NULL
```

Comment:
This function does really find anything, it's more if the identifier match's the handle's identifier then the handle's function is returned.